client/src/pages/AccountSettingsPage.jsx

```diff
@@ -0,0 +1,320 @@
+import {
+  Button,
+  Card,
+  Grid,
+  Stack,
+  TextField,
+  Typography,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+
+import BackgroundShapes from '../components/BackgroundShapes';
+import CloseAccountButton from '../components/CloseAccountButton';
+import ErrorMessage from '../components/ErrorMessage';
+import SignOutButton from '../components/SignOutButton';
+import {
+  getUserAccountInfo,
+  updateUserEmail,
+  updateUserName,
+  updateUserPassword,
+} from '../utils/api';
+import { isEmailValid, isPasswordValid } from '../utils/helper';
+import BasePage from './BasePage';
+
+const AccountSettingsPage = () => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [email, setEmail] = useState('');
+  const [confirmEmail, setConfirmEmail] = useState('');
+  const [errorMessage, setErrorMessage] = useState(null);
+
+  const [editableFields, setEditableFields] = useState({
+    name: false,
+    password: false,
+    email: false,
+  });
+
+  const rowFields = [
+    {
+      editableFieldKey: 'name',
+      previewModeFields: {
+        label: 'Name',
+        value: firstName + ' ' + lastName,
+        buttonText: 'Change name',
+      },
+      editModeFields: {
+        fields: [
+          {
+            label: 'First name',
+            value: firstName,
+            isHidden: false,
+            onChange: (event) => setFirstName(event.target.value),
+          },
+          {
+            label: 'Last name',
+            value: lastName,
+            isHidden: false,
+            onChange: (event) => setLastName(event.target.value),
+          },
+        ],
+        buttonText: 'Save changes',
+        getErrorMessage: async () => {
```

Comment:
nit: I feel like these functions should be called something else since the main action they're trying to do is update user data, not rly return an error message. Maybe something like setUserData() or something. That's just my opinion tho, feel free to disagree